electricitymap/contrib/lib/models/events.py

```diff
@@ -119,11 +154,31 @@ def set_value(
         If correct_negative_with_zero is set to True, negative values will be set to 0 instead of None.
         This method keeps track of values that have been corrected.
         """
-        if correct_negative_with_zero and value is not None and value < 0:
-            value = 0
-            self._corrected_negative_values.add(mode)
+        value = self._correct_negative_value(mode, value, correct_negative_with_zero)
         self.__setattr__(mode, value)
 
+    def add_value(
+        self,
+        mode: str,
+        value: Optional[float],
+        correct_negative_with_zero: bool = False,
+    ) -> None:
+        """Adds the provided value to the existing value of the provided mode. Negative values are set to None by default.
+        If correct_negative_with_zero is set to True, negative values will be set to 0 instead of None.
+        This method keeps track of modes that have been corrected.
+        """
+        value = self._correct_negative_value(mode, value, correct_negative_with_zero)
+        if value is None:
```

Comment:
```suggestion
        super().add_value(mode, value)
```
In that case I suggest doing this and remove the code below :)

Reply:
Oh I have not really used that approach before, seems useful!

I'll test it!